### PR TITLE
Seggregate ETH

### PR DESCRIPTION
**Pull request by alekece**
This PR contains the entire ethereum/ERC20 seggregation.

**I finally made it !**

giphy

### CODE REVIEW

File: tools/lc [hunk 1]
```diff
@@ -25,23 +25,23 @@ function cmd_project_new {
...
 
   echo "—> symlinking ledger-core’s libraries"
   ln -sf $PWD/ledger-core/lib $project_name/lib
   ln -sf $PWD/ledger-core/cmake $project_name/cmake
 
   echo "—> generating CMake configuration"
-  sed s/\$project_name/$project_name/g ./tools/templates/CMakeLists-project.txt.tpl > $project_name/CMakeLists.txt
-  sed s/\$project_name/$project_name/g ./tools/templates/CMakeLists-project-src.txt.tpl > $project_name/src/CMakeLists.txt
+  sed -e s/\$project_name/$project_name/g ./tools/templates/CMakeLists-project.txt.tpl > $project_name/CMakeLists.txt
+  sed -e s/\$project_name/$project_name/g -e s/\$coin_name/$coin_name/g ./tools/templates/CMakeLists-project-src.txt.tpl > $project_name/src/CMakeLists.txt
```
Comment (phaazon) [inline tools/lc]: This should have been in a different PR, because now I’m going to have issues with XRP.

File: tools/lc [hunk 2]
```diff
@@ -25,23 +25,23 @@ function cmd_project_new {
...
 
   echo "—> generating IDL entry point"
   cp ./tools/templates/idl.djinni $project_name/idl/
 
   echo "—> creating migrations files"
-  sed s/\$project_name/$coin_name/g ./tools/templates/migrations.hpp > $project_name/inc/database/migrations.hpp
-  sed s/\$project_name/$coin_name/g ./tools/templates/migrations.cpp > $project_name/src/database/migrations.cpp
+  sed s/\$project_name/$coin_name/g ./tools/templates/Migrations.hpp > $project_name/inc/$coin_name/database/Migrations.hpp
```
Comment (phaazon) [inline tools/lc]: Ditto.

File: ledger-core-ethereum/inc/ethereum/database/Migrations.hpp [hunk 3]
```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+#include <core/database/Migrations.hpp>
+
+namespace ledger {
+    namespace core {
+        /// Tag type.
+        struct ethereumMigration {
```
Comment (phaazon) [inline ledger-core-ethereum/inc/ethereum/database/Migrations.hpp]: Please update the typename here.

File: ledger-core-ethereum/src/ethereum/database/Migrations.cpp [hunk 4]
```diff
@@ -0,0 +1,112 @@
+#include <ethereum/database/Migrations.hpp>
+
+namespace ledger {
+    namespace core {
+        int const ethereumMigration::coinID = 60;
```
Comment (phaazon) [inline ledger-core-ethereum/src/ethereum/database/Migrations.cpp]: Don’t forget the typename. :)

File: ledger-core-ethereum/idl/idl.djinni [hunk 5]
```diff
@@ -0,0 +1,157 @@
+@extern "../../ledger-core/idl/core.yaml"
+
+@import "addresses.djinni"
+@import "erc20.djinni"
+
+# An Ethereum-like wallet.
+EthereumLikeWallet = interface +c {
```
[the remark this answers is not in the document]
Comment (alekece) [reply ledger-core-ethereum/idl/idl.djinni]: You got a point - this interface is never used in the ETH code so I simply removed it.

**Pull request merged**